core/coreapi/interface/swarm.go

```diff
@@ -0,0 +1,37 @@
+package iface
+
+import (
+	"context"
+	"time"
+
+	peer "gx/ipfs/QmQsErDt8Qgw1XrsXf2BpEzDgGWtB1YLsTAARBup5b6B9W/go-libp2p-peer"
+	ma "gx/ipfs/QmYmsdtJ3HsodkePE3eU3TsCaP2YvPZJ4LoXnNkDE5Tpt7/go-multiaddr"
+)
+
+// PeerInfo contains information about a peer
+type PeerInfo interface {
```

Comment:
Is this `PeerInfo` or `ConnectionInfo`? It looks like this is a per-connection datastructure.